domain/bakerystorage/state/types.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"database/sql/driver"
+
+	"github.com/go-macaroon-bakery/macaroon-bakery/v3/bakery"
+	"github.com/juju/errors"
+)
+
+type Config struct {
+	LocalUsersPrivateKey              *keyScanner `db:"local_users_private_key"`
+	LocalUsersPublicKey               *keyScanner `db:"local_users_public_key"`
+	LocalUsersThirdPartyPrivateKey    *keyScanner `db:"local_users_third_party_private_key"`
+	LocalUsersThirdPartyPublicKey     *keyScanner `db:"local_users_third_party_public_key"`
+	ExternalUsersThirdPartyPrivateKey *keyScanner `db:"external_users_third_party_private_key"`
+	ExternalUsersThirdPartyPublicKey  *keyScanner `db:"external_users_third_party_public_key"`
+	OffersThirdPartyPrivateKey        *keyScanner `db:"offers_third_party_private_key"`
+	OffersThirdPartyPublicKey         *keyScanner `db:"offers_third_party_public_key"`
+}
+
+type keyScanner struct {
+	key bakery.Key
+}
+
+// Key returns the keyScanner's key
```

Comment:
Probably also point out this implements the valuer interface.

Reply:
That is done on line 47.

On 2nd thought, this method `.Key()` isn't helpful so should probably go

Reply:
Ah yea, brain fart